enhancing/enhancer.js

```diff
@@ -6,17 +6,37 @@ module.exports = {
 };
 
 function succeed(item) {
-  return { ...item };
+  if (typeof item !== 'object' || item == null || Array.isArray(item))
+    return undefined;
+  const enhancement =
+    item.enhancement < 20 ? item.enhancement + 1 : item.enhancement;
+  return {
+    ...item,
+    enhancement,
+  };
 }
 
 function fail(item) {
-  return { ...item };
+  if (typeof item !== 'object' || item == null || Array.isArray(item))
+    return undefined;
+  const enhancement =
+    item.enhancement > 16 ? item.enhancement - 1 : item.enhancement;
+  const durability =
+    item.enhancement >= 15 ? item.durability - 10 : item.durability - 5;
```

Comment:
I look at this and have to wonder if there's a better/shorter way to go about this... after thinking about it for a minute, I'd say the following might be a good start to reduce the unnecessary repetition:
```javascript
const enhancement = item.enchancement - (item.enhancement > 16 ?  1 : 0);
const durability = item.durability - (item.enhancement >= 15 ? 10 : 5);
return { ...item, enhancement, durability };
```
A step further in the right direction could be object destructuring:
```javascript
let { enhancement, durability } = item;
enhancement -= enhancement > 16 ? 1 : 0;
durability -= item.enhancement >= 15 ? 10 : 5;
return { ...item, enhancement, durability };
```